packages/devtools_app/lib/src/scaffold.dart

```diff
@@ -260,11 +251,15 @@ class DevToolsScaffoldState extends State<DevToolsScaffold>
       // We want to avoid this because the routes underneath the existing
       // '/snapshot' route should remain unchanged while '/snapshot' sits on
       // top.
-      if (ModalRoute.of(context).settings.name == snapshotRoute) {
-        Navigator.popAndPushNamed(context, snapshotRoute, arguments: args);
+      final routerDelegate = DevToolsRouterDelegate.of(context);
+      if (routerDelegate.currentConfiguration.page == snapshotPageId) {
+        routerDelegate
+          ..popPage()
```

Comment:
is the pop page necessary here? one way to test would be to drag-and-drop to import a file multiple times in a row without clicking "Exit offline mode"? Then if you click "Exit offline mode", the router should return to the landing screen as expected.

Reply:
Based on the comment above, I believe it was to avoid adding more and more snapshot pages to the navigation stack when dragging on many.

I've changed this slightly, adding a "replacePage" method, as the previous one would call `notifyListeners()` twice (once after pop, once after push). The Exit button works correctly, however the back button isn't working right - still working on understanding why.